.github/workflows/deploy.yml

```diff
@@ -188,33 +183,52 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v3
-      - name: NPM Install
-        uses: bahmutov/npm-install@v1
+
+      - name: Setup Node v16 for Yarn v3
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16.15.0' # Current LTS version
+
+      - name: Enable Corepack for Yarn v3
+        run: corepack enable
+
+      - name: Install Yarn v3
+        uses: borales/actions-yarn@v3
+        with:
+          cmd: set version stable
+
+      - name: Install Dependencies
+        uses: borales/actions-yarn@v3
+        env:
+          YARN_ENABLE_IMMUTABLE_INSTALLS: false
+        with:
+          cmd: install
+
       - name: Gatsby Cache
         uses: actions/cache@v2
         with:
           path: |
             public
             .cache
-          key:
-            ${{ needs.set-state.outputs.branch_short_ref }}-gatsby-cache-${{
-            github.run_id }}
+          key: ${{ needs.set-state.outputs.branch_short_ref }}-gatsby-cache-${{ github.run_id }}
```

Comment:
Some of these steps are the same for both `build-stage` and `build-production`. Can you extract them in separate job(s) and reuse them?
For e.g.`install-dependencies`, `cache-setup`

Reply:
Yes but we would be reluctant to do so since it creates more file to keep track off as we push platform wide updates. 😓 We already have the `path_prefix` script which we missed on a few repos and had to circle back for.

Reply:
I think even separate jobs in current file would help with reducing the duplicated code and improve the maintenance aspect